pkgs/development/compilers/llvm/15/llvm/gnu-install-dirs-polly.patch

```diff
@@ -0,0 +1,102 @@
+diff --git a/tools/polly/CMakeLists.txt b/tools/polly/CMakeLists.txt
```

Comment:
Was this patch tested?

There's no path to get to an llvm15 with this included afaics in this PR:

```
$ nix search github:rrbutani/nixpkgs/feature/llvm-15 polly | grep 15
# empty
```

There are packages for older llvm versions if you take out the grep.

Reply:
Good catch!

The patch is only pulled in with `enablePolly = true` on `llvmPackages_15.llvm` but it's actually no longer needed (obviated by [patches @Ericson2314 sent in](https://reviews.llvm.org/D117541) that [made it into LLVM 14](https://github.com/llvm/llvm-project/commit/d3b756c51c1b42f653b8abf2b2e83a2321befb9d)).

----

I did some digging:
  - polly was [initially added](https://github.com/NixOS/nixpkgs/commit/ed60483257b62c32b5b58c6e91c9c8cd586f77d0) to `llvmPackages_7`, disabled by default but with polly-enabled `llvm` and `clang` package variants added to the package set: #55065
  - this made it into [`llvmPackages_8`](https://github.com/NixOS/nixpkgs/blob/3c3d6bedf040a42096959502b9a608568e105149/pkgs/development/compilers/llvm/8/default.nix#L83) (#56645) and [`llvm_packages_9`](https://github.com/NixOS/nixpkgs/blob/3c3d6bedf040a42096959502b9a608568e105149/pkgs/development/compilers/llvm/9/default.nix#L83) (#69115) (via `llvmPackages_7`, 8 was copied from 7, 9 from 8)
  - `llvmPackages_10` initially inherited this setup (`enablePolly` set to false, `llvm` and `clang` package variants) but later [switched to enabling polly by default and doing away with the polly-enabled packages](https://github.com/NixOS/nixpkgs/commit/e9aa8770ea8bd872da0d04cb98a7856ba1fc5412) in #82781
  - in #94204 `llvmPackages_11` [initially inherited the above](https://github.com/NixOS/nixpkgs/commit/28f2797a81e738234cdc57a688afb136f692672b#diff-860ddef36193cf2cd6e4ca65c1dccba9be723417d043d765f7cceff4dd179740R21) (polly enabled by default) but then [disabled polly](https://github.com/domenkozar/nixpkgs/blame/17a8c9c4bfd9042b92b1d07d638a3dc1157b11d4/pkgs/development/compilers/llvm/11/llvm/default.nix#L21) (note the TODO in the commit message and [this comment](https://github.com/NixOS/nixpkgs/pull/94204#issuecomment-676413893))
     + later, #125184 [added](https://github.com/NixOS/nixpkgs/pull/125184/commits/9e6417f2a4e0431dc1c9cb86bd38161c4f0f53a2) the polly-enabled package variants that were removed in `llvmPackages_10` to `llvmPackages_11` (clang *and* LLVM variants)
  - #111487 added `enablePolly` – disabled by default – and polly enabled `llvm`/`libllvm` package variants to [`llvmPackages_5`](https://github.com/NixOS/nixpkgs/blame/7869d1654517c028aa76fc1dedc9b5ac301a867a/pkgs/development/compilers/llvm/5/llvm/default.nix#L18) and [`llvmPackages_6`](https://github.com/NixOS/nixpkgs/blame/7869d1654517c028aa76fc1dedc9b5ac301a867a/pkgs/development/compilers/llvm/6/llvm/default.nix#L18) but *without* the polly enabled `clang` package that `llvmPackages_7`+ had
    + this PR also added this TODO on `llvmPackages_11` saying to `enablePolly` by default: https://github.com/NixOS/nixpkgs/blob/7869d1654517c028aa76fc1dedc9b5ac301a867a/pkgs/development/compilers/llvm/11/llvm/default.nix#L22
     * (but did not add a similar TODO to `llvmPackages_12`, the latest at the time)
  - [`llvmPackages_12`](https://github.com/NixOS/nixpkgs/blame/3c3d6bedf040a42096959502b9a608568e105149/pkgs/development/compilers/llvm/12/llvm/default.nix#L22) was copied from `llvmPackages_11`
     + Can't find a PR for this one but it's dated April 1st, 2021 which puts it:
       * after #94204 (August 25th, 2020; the PR that introduced `llvmPackages_11` with polly disabled *and* with no polly-enabled package variants)
       * but before #111487 (April 30th, 2021; the PR that added a TODO for `enablePolly`)
       * and before #125184 (June 3rd, 2021; the PR that added polly-enabled package variants back)
  - in #120780 `llvmPackages_git` was introduced, copying from `llvmPackages_12` at this point
  - [`llvmPackages_13`](https://github.com/NixOS/nixpkgs/blame/3c3d6bedf040a42096959502b9a608568e105149/pkgs/development/compilers/llvm/13/llvm/default.nix#L23) was added in #132643, copying from `llvmPackages_git` and [`llvmPackages_14`](https://github.com/NixOS/nixpkgs/blame/3c3d6bedf040a42096959502b9a608568e105149/pkgs/development/compilers/llvm/14/llvm/default.nix#L24) in #162104, again copying from `llvmPackages_git`

----

The TL;DR is that because of the timing of the commit that introduced `llvmPackages_12` (i.e. after polly was disabled in `llvmPackages_11` but before a TODO was added and before polly enabled package variants were reintroduced), we haven't been testing polly support since `llvmPackages_12`.

----

@LunNova thanks for catching this! I'll remove this patch file and try re-enabling polly by default (hopefully whatever issue that led us to disable it in the first place has since resolved itself).

Outside of the scope of this PR but: I'm fairly certain `enablePolly` is broken in `llvmPackages_14` (the patch won't apply cleanly) and likely in `llvmPackages_{12,13}` too; I'll add a note to the OP to fix this in a follow up PR.